public/js/mage/adminhtml/form.js

```diff
@@ -436,135 +437,214 @@ SelectUpdater.prototype = {
     }
 };
 
-
 /**
- * Observer that watches for dependent form elements
- * If an element depends on 1 or more of other elements, it should show up only when all of them gain specified values
+ * Observer that watches for dependent form elements with support for complex conditions
  */
-FormElementDependenceController = Class.create();
-FormElementDependenceController.prototype = {
+class FormElementDependenceController {
+    static MODE_NOT = 'NOT';
+    static MODE_AND = 'AND';
+    static MODE_OR  = 'OR';
+    static MODE_XOR = 'XOR';
+
     /**
-     * Structure of elements: {
-     *     'id_of_dependent_element' : {
-     *         'id_of_master_element_1' : 'reference_value',
-     *         'id_of_master_element_2' : 'reference_value'
-     *         'id_of_master_element_3' : ['reference_value1', 'reference_value2']
-     *         ...
-     *     }
-     * }
-     * @param object elementsMap
-     * @param object config
+     * @param {Object.<string, Object>} elementsMap - key/value pairs of target fields and their conditions to be visible
+     * @param {Object} [config] - config options
+     * @param {Object.<string, string>} [config.field_map] - key/value pairs of field aliases to their associated DOM IDs.
+     * @param {Object.<string, string>} [config.field_values] - key/value pairs of fallback values for fields not present in the form
+     * @param {number} [config.levels_up] - deprecated: the number of ancestor elements to find the parent element to hide
+     * @param {boolean} [config.can_edit_price] - deprecated: prevent enabling price inputs, only use this option if dependence block contains no other elements!
      */
-    initialize : function (elementsMap, config)
-    {
-        if (config) {
-            this._config = config;
+    constructor(elementsMap, config = {}) {
+        console.log(elementsMap, config);
```

Comment:
```suggestion
```

Reply:
That line is in the test module commit that will be reverted :)